external/storm-submit/src/main/java/org/apache/storm/submit/dependency/AetherUtils.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.storm.submit.dependency;
+
+import org.sonatype.aether.artifact.Artifact;
+import org.sonatype.aether.graph.Dependency;
+import org.sonatype.aether.graph.Exclusion;
+import org.sonatype.aether.util.artifact.DefaultArtifact;
+import org.sonatype.aether.util.artifact.JavaScopes;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+public class AetherUtils {
+    private AetherUtils() {
+    }
+
+    public static Dependency parseDependency(String dependency) {
+        List<String> dependencyAndExclusions = Arrays.asList(dependency.split("!"));
```

Comment:
'!' occasionally represents 'not', so I think it would be easier to be recognized as excluded. Do we feel similar from '~'?

Refer to http://www.gnu.org/software/bash/manual/bashref.html#Quoting we should avoid to use '$', '`', '\', '"', '!', '@', '*' in double quotes if we want to avoid escape.